src/flint/fmpz.jl

```diff
@@ -1322,60 +1326,78 @@ function binom(n::Int, k::Int)
 end
 
 @doc Markdown.doc"""
-    moebiusmu(x::fmpz)
+    moebius_mu(x::fmpz)
 > Returns the Moebius mu function of $x$ as an `Int`. The value
 > returned is either $-1$, $0$ or $1$. If $x < 0$ we throw a `DomainError()`.
 """
-function moebiusmu(x::fmpz)
+function moebius_mu(x::fmpz)
    x < 0 && throw(DomainError("Argument must be non-negative: $x"))
    return Int(ccall((:fmpz_moebius_mu, :libflint), Cint,
                     (Ref{fmpz},), x))
 end
 
+moebius_mu(x::Integer) = moebius_mu(fmpz(x))
+
 @doc Markdown.doc"""
-    jacobi(x::fmpz, y::fmpz)
+    jacobi_symbol(x::fmpz, y::fmpz)
 > Return the value of the Jacobi symbol $\left(\frac{x}{y}\right)$. If
-> $y \leq x$ or $x < 0$, we throw a `DomainError()`.
+> $y \leq 0$, we throw a `DomainError()`.
 """
-function jacobi(x::fmpz, y::fmpz)
-   y <= x && throw(DomainError("Second argument must be larger than first argument: $x $y"))
-   x < 0 && throw(DomainError("First argument must be non-negative: $x"))
+function jacobi_symbol(x::fmpz, y::fmpz)
+   y <= 0 && throw(DomainError(y, "Modulus must be positive"))
+   if x < 0 || x >= y
+      x = mod(x, y)
+   end
    return Int(ccall((:fmpz_jacobi, :libflint), Cint,
                     (Ref{fmpz}, Ref{fmpz}), x, y))
 end
 
+function jacobi_symbol(x::Int, y::Int)
+   y <= 0 && throw(DomainError(y, "Modulus must be positive"))
+   if x < 0 || x >= y
+      x = mod(x, y)
+   end
+   return Int(ccall((:n_jacobi, :libflint), Cint, (Int, UInt), x, UInt(y)))
+end
+   
+jacobi_symbol(x::Integer, y::Integer) = jacobi_symbol(fmpz(x), fmpz(y))
+
 @doc Markdown.doc"""
-    sigma(x::fmpz, y::Int)
+    divisor_sigma(x::fmpz, y::Int)
 > Return the value of the sigma function, i.e. $\sum_{0 < d \;| x} d^y$. If
 > $y < 0$ we throw a `DomainError()`.
 """
-function sigma(x::fmpz, y::Int)
+function divisor_sigma(x::fmpz, y::Int)
    y < 0 && throw(DomainError("Second argument must be non-negative: $y"))
    z = fmpz()
    ccall((:fmpz_divisor_sigma, :libflint), Nothing,
          (Ref{fmpz}, Ref{fmpz}, Int), z, x, y)
    return z
 end
 
+divisor_sigma(x::Integer, y::Int) = divisor_sigma(fmpz(x), y)
```

Comment:
What about `divisor_sigma(x::Integer, y::Integer) = divisor_sigma(fmpz(x), Int(y))`? (just a suggestion, the question of widening in general is still open and can be addressed later, there is also an issue open in AbstractAlgebra.jl).

Reply:
Also add test.

Reply:
There are hundreds of these functions that accept an Int that could be widened to Integer. Honestly, I really don't care about this, except to say that I really don't like the idea of allowing BigInt, since the documentation will then reflect that any Integer is allowed, and people may get a surprise when they find out it is not. Someone else can fix this if they think they have time.

I'll add tests if I have time today (I probably won't).

Reply:
Ah OK, let's keep it like this for now. If you like I can add tests. I wouldn't mind adding the tests (to your branch) if you like.

Reply:
That would be helpful. I got nowhere near getting everything done today.

Reply:
Actually it seems I can't add stuff to your branch. I can either push tests to another branch which you merge into your branch, or I can open a separate PR after this one is merged (maybe the simplest).

Reply:
Yeah just do that. These aren't going to break.

This is what I've been working on. All the examples are broken until we get all these PR's merged. But I'm moderately happy with the result so far.

https://oscar-system.github.io/Oscar.jl/latest/integer.html

Reply:
Looks good!